File: design-pattern/singalton-pattern/singalton.ts

class Singalton {
    static instance: Singalton | null = null;

    constructor() {
        console.log('this is singalton');
    };

    static getInstance(): Singalton {
        if (this.instance == null) {
            this.instance = new Singalton();
        }

        return this.instance;
    };
};

const singalton1 = Singalton.getInstance();
const singalton2 = Singalton.getInstance();

if (singalton1 == singalton2) {
    console.log("both are same");
} else {
    console.log("both are different")
}

File: design-pattern/singalton-pattern/singalton.cpp
#include <iostream>

using namespace std;

class Singalton
{
private:
    static Singalton *instance;

    Singalton()
    {
        cout << "this is singalton" << endl;
    };

public:
    static Singalton *getInstance()
    {
        if (instance == nullptr)
        {
            instance = new Singalton();
        }

        return instance;
    }
};

Singalton *Singalton::instance = nullptr;

int main(void)
{
    Singalton *Singalton1 = Singalton::getInstance();
    Singalton *Singalton2 = Singalton::getInstance();

    if (Singalton1 == Singalton2)
    {
        cout << "Both instances are the same" << endl;
    }
    else
    {
        cout << "Instance are difference" << endl;
    };

    return 0;
}